demo/api/src/products/custom-product.resolver.ts

```diff
@@ -1,18 +1,45 @@
 import { RequiredPermission } from "@comet/cms-api";
 import { InjectRepository } from "@mikro-orm/nestjs";
 import { EntityRepository } from "@mikro-orm/postgresql";
-import { Mutation, Resolver } from "@nestjs/graphql";
+import { Mutation, Query, Resolver } from "@nestjs/graphql";
+import { getMessages } from "@src/common/lang";
+import { createIntl, createIntlCache, IntlCache } from "react-intl";
 
 import { Product, ProductStatus } from "./entities/product.entity";
 
 @Resolver(() => Product)
 @RequiredPermission(["products"], { skipScopeCheck: true })
 export class CustomProductResolver {
-    constructor(@InjectRepository(Product) private readonly repository: EntityRepository<Product>) {}
+    private readonly intlCache: IntlCache;
+    constructor(@InjectRepository(Product) private readonly repository: EntityRepository<Product>) {
+        this.intlCache = createIntlCache();
+    }
 
     @Mutation(() => Boolean)
     async publishAllProducts(): Promise<boolean> {
         await this.repository.nativeUpdate({ status: { $ne: ProductStatus.Published } }, { status: ProductStatus.Published });
         return true;
     }
+
+    @Query(() => String)
+    async myTranslatedString(): Promise<string> {
+        const language = "de"; // TODO correctly set the language, maybe move into the mail template
+        const messages = getMessages(language);
+
+        const intl = createIntl(
```

Comment:
Having to create the intl object every time we want to mask a messages seems cumbersome to me. Should we create an IntlModule with a service for this?

Reply:
Yes, I think this is a good idea. Probably directly provided by app-module or as a global service provided by config-module? I don't know if this could be a candidate for a comet-module...

Reply:
created a translation-service in the config-module 👍

Reply:
I wasn't aware that this will be only used for mail rendering with react-intl. Then having a service doesn't make sense, right?

Reply:
I don't know. A service for something like translation-strings did sound reasonable for me, because this can be used for mailing but also for generating pdfs. and maybe even to provide some default-values for entities generated without direct user input.